File: app/src/main/java/com/example/madarsofttask/ui/list/ListUsersViewModel.java
package com.example.madarsofttask.ui.list;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.example.madarsofttask.data.models.User;
import com.example.madarsofttask.data.repositories.UserRepository;

import java.util.List;


public class ListUsersViewModel extends ViewModel {

    private LiveData<List<User>> allUsers;

    public ListUsersViewModel(){
        UserRepository myUserRepo = UserRepository.getInstance();
        allUsers = myUserRepo.getUsers();
    }

    LiveData<List<User>> getAllUsers() {
        System.out.println("Users are : "+allUsers.toString());
        return allUsers;
    }
}

File: app/src/main/java/com/example/madarsofttask/ui/list/ListUsersViewModel.kt
package com.example.madarsofttask.ui.list

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.madarsofttask.data.models.User
import com.example.madarsofttask.data.repositories.UserRepository.Companion.instance

class ListUsersViewModel : ViewModel() {
    val allUsers: MutableLiveData<List<User?>?>?

    init {
        val myUserRepo = instance
        allUsers = myUserRepo?.users
    }
}